src/python/pants/testutil/rule_runner.py

```diff
@@ -511,6 +511,15 @@ def write_files(
             )
         return tuple(paths)
 
+    def read_file(self, file: str | PurePath, mode: str = "r") -> str | bytes:
+        """Read a file that was written to the build root, useful for testing."""
+        path = os.path.join(self.build_root, file)
+        with safe_open(path, mode=mode) as fp:
+            if "b" in mode:
+                return bytes(fp.read())
+            else:
+                return str(fp.read())
```

Comment:
fair, I'll flatten the `else`
Without the else, it reads to me more as a guard clause, rather than 2 equal branches. I think of it as inlining the intermediate var `content` in the following
```python
if "b" in mode:
    content = bytes(fp.read())
else:
    content = bytes(fp.read())
return content
```